### PR TITLE
docs(README): add virtual node documentation

### PR DESCRIPTION
**Pull request by TylorS**
Add documentation for the structure and properties of a virtual
node used by snabbdom and created by h().

### CODE REVIEW

**Comment by TylorS** [comment]
Sorry it took so long from when I wrote this in #78. Please let me know if there are any things I can adjust to make this suitable for merging :)

File: README.md [hunk 1]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+ - [sel](#sel--string)
+ - [data](#data--object)
+ - [children](#children--array)
+ - [text](#text--string)
+ - [elm](#elm--element)
+ - [key](#key--string--number)
+
+#### sel : String
+The `.sel` property of a virtual node is the CSS selector passed to [`h()`](#snabbdomh) during creation.
+For example: `h('div#container, {}, [...])` will create a a virtual node which has `div#container` as it's `.sel` property.
```
Comment (foxdonut) [inline README.md]: missing closing quote: should be `h('div#container',`

File: README.md [hunk 2]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+
+#### data : Object
+The `.data` property of a virtual node is the place to add information for [modules](#modules-documentation) to access and manipulate the real DOM element when it is created; Add styles, CSS classes, attributes, etc.
+
+The data object is the (optional) second parameter to [`h()`](#snabbdomh)
+
+For example `h('div', {props: {className: 'container'}, [...]}` will produce a virtual node with
```
Comment (foxdonut) [inline README.md]: I think this should be `h('div', {props: {className: 'container'}}, [...])`

File: README.md [hunk 3]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+```js
+{
+  "props": {
+    className: "container"
+  }
+}
+```
+as it's `.data` object.
```
Comment (foxdonut) [inline README.md]: as **its**

File: README.md [hunk 4]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+
+#### children : Array<VNode>
+The `.children` property of a virtual node is the third (optional) parameter to [`h()`](#snabbdomh) during creation.
+`.children` is simply and Array of virtual nodes that should be added as children of the parent DOM node upon creation.
```
Comment (foxdonut) [inline README.md]: is simply **an** Array

File: README.md [hunk 5]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+   sel: 'h1',
+   data: {},
+   children: undefined,
+   text: 'Hello, World',
+   elm: Element,
+   key: undefined,
+ }
+]
+```
+as it's `.children` property.
```
Comment (foxdonut) [inline README.md]: as **its**

File: README.md [hunk 6]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+
+#### text : string
+The `.text` property is created when the a virtual node is created with only a single child that possesses text and only requires `document.createTextNode()` to be used.
```
Comment (foxdonut) [inline README.md]: when **the a** should be either 'the' or 'a'

File: README.md [hunk 7]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+
+For example: `h('h1', {}, 'Hello')` will create a virtual node with `Hello` as it's `.text` property.
```
Comment (foxdonut) [inline README.md]: as its

File: README.md [hunk 8]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+
+#### elm : Element
+
+The `.elm` property of a virtual node is a pointer to the real DOM node created by snabbdom. This property is very useful to do calculations in [hooks](#hooks) as well as [modules](#modules-documentation).
+
+#### key : string | number
+
+The `.key` property is created when a key is provided inside of your [`.data`](#data--object) object. The `.key` property is used to keep pointers to DOM nodes that existed previously to avoid recreating them if it is unnecessary. This is very useful for things like list reordering. A key much be either a string or a number to allow for proper lookup as it is stored internally as a key value pair inside of an object, where `.key` is the key and the value is the [`.elm`](#elm--element) property created.
```
Comment (foxdonut) [inline README.md]: A key -much- must be either

**Comment by thurt** [comment]
This is very helpful information which reveals important (yet undocumented) features--especially `data.key`. 

I would like to see more documented recommendations for how to use snabbdom for specific purposes. I mention this because this PR has some "recommendation" style language that is very helpful to me (and I am sure it will be to others).

File: README.md [hunk 9]
```diff
@@ -491,6 +492,69 @@ The view function here is only an example. In practice thunks are only
...
+ - [sel](#sel--string)
+ - [data](#data--object)
+ - [children](#children--array)
+ - [text](#text--string)
+ - [elm](#elm--element)
+ - [key](#key--string--number)
+
+#### sel : String
+The `.sel` property of a virtual node is the CSS selector passed to [`h()`](#snabbdomh) during creation.
+For example: `h('div#container', {}, [...])` will create a a virtual node which has `div#container` as it's `.sel` property.
```
Comment (foxdonut) [inline README.md]: Don't forget to change this "it's" to "its" :)

**Comment by paldepind** [comment]
@TylorS

> Sorry it took so long from when I wrote this in #78.

No problem at all. Thank you very much for this PR. I think it documents some important things very nicely :+1: 

@thurt, I can see what you mean. The current documentation is mostly just a list of what one can do. I think the reason for that is that the intention behind Snabbdom is that people can build applications in any way they want. It tries to be an unopinionated virtual DOM library. But this certainly makes it harder to figure out what to do with Snabbdom.

**Pull request merged**